tools/wake-unit/filepath.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <wcl/filepath.h>
+#include <wcl/xoshiro_256.h>
+
+#include <random>
+#include <string>
+#include <vector>
+
+#include "unit.h"
+
+static std::vector<std::string> to_vec(std::string str) {
+  std::vector<std::string> actual;
+
+  for (std::string&& node : wcl::make_filepath_range(str)) {
+    actual.emplace_back(std::move(node));
+  }
+
+  return actual;
+}
+
+#define EXPECT_EQUAL_VEC(expected, actual)          \
```

Comment:
Can we lift this into `unit.h`?

Reply:
Yeah I can make a better version honestly that just specializes to vectors instead of this bodge.